### DISCUSSION TITLE
Allow collaboration on draft records

### DISCUSSION THREAD

**Discussion by edbennett** [inveniosoftware/product-rdm#201] [Ideas]
Currently (unless I have overlooked something) records have to be prepared by a single user, which can be a bottleneck for larger/more complex records.

It would be useful if it were possible for multiple authors to collaborate on preparing a draft record before submission.

I imagine the simplest way to achieve this would be to add a check as to whether a user with an associated ORCID has that ORCID listed as an author on the draft, and if so grant access. The original creator of the draft could then share the link to the draft with collaborators via email/instant messaging.

(An invite/link sharing system similar to published records might be easier for some, but would also introduce more UI complexity.)

**Comment by anikachurilova** [comment]
Thank you for the question!
The behaviour you are describing can be achieved by creating an access link with "edit" access level. The created link will have a token that lets any user that you share the link with to be able to edit your record! As a record owner, you can then modify the access level or delete the link when needed.
To find this feature, please navigate to the "Links" tab of a "Share access" modal
![Screenshot 2024-04-25 at 16 38 39](https://github.com/inveniosoftware/product-rdm/assets/61321254/72c28205-930c-445e-9ce5-47fcfefaa66c)

PS: the feature of granting permissions to a specific person or group is coming 🚀

**Comment by edbennett** [comment]
Thanks Anika. The record has to have been created for that to be the case though, right? What I was looking for was for a similar kind of collaboration but while it is still in Draft status, before the first version is made public.

**Reply by tmorrell** [reply]
Yup, the share links won't work for that use case. However there are other options.

Currently you can submit a record to a community. If all members who want to collaborate are part of that community, they will have access to the draft record. 

You could also publish a fully restricted record, and then use the share link to provide access for editing.

There is also work at the moment to provide individual/group permissions for individual records, which is probably exactly what you're looking for https://github.com/inveniosoftware/product-rdm/issues/198

**Reply by edbennett** [reply]
Aha, the second bullet is exactly what I was looking for—I searched but didn't spot that one. Thanks!

**Comment by anikachurilova** [comment]
I see now, thank you for the clarification!

Indeed, right now we are in progress of working on adding the Share button to the upload creation form, so that it is possible to share the record in the draft state. It will come soon 🚀

**Comment by lnielsen** [comment]
I'm closing the discussion since it's in progress and on the roadmap - https://github.com/inveniosoftware/product-rdm/issues/198 
Feel free to reopen in case the desired approach is the one of using the ORCID to grant access.

@anikachurilova Perhaps you can post a couple of screenshots on the mentioned issue of how you can share a draft record with what's being developed.
